recursion/factorial.py

```diff
@@ -19,3 +19,19 @@
 # Ex:
 #   factorial(5) # return 120
 #   factorial(6) # return 720
+def factorial(number: int) -> int:
+    if number < 0:
+        return -1
+
+    memo = [None] * (number + 1)
+    memo[0] = 1
+
+    def factorialHelper(number: int) -> int:
+        if memo[number] is not None:
```

Comment:
I told you that if memo[sum] is None, then memo[sum] is False otherwise it is True. You don’t need to check if it is not None. Just:

    if memo[sum]:
        <code block>